program/src/state.rs

```diff
@@ -711,27 +711,52 @@ impl Validator {
 
     pub fn check_can_be_removed(&self) -> Result<(), LidoError> {
         if self.active {
-            msg!("Refusing to remove validator because it is still active, deactivate it first.");
             return Err(LidoError::ValidatorIsStillActive);
         }
         if self.fee_credit != StLamports(0) {
-            msg!(
-                "Validator still has tokens to claim. Reclaim tokens before removing the validator"
-            );
             return Err(LidoError::ValidatorHasUnclaimedCredit);
         }
         if self.has_stake_accounts() {
-            msg!("Refusing to remove validator because it still has stake accounts, unstake them first.");
             return Err(LidoError::ValidatorShouldHaveNoStakeAccounts);
         }
         if self.has_unstake_accounts() {
-            msg!("Refusing to remove validator because it still has unstake accounts, withdraw them first.");
             return Err(LidoError::ValidatorShouldHaveNoUnstakeAccounts);
         }
         // If not, this is a bug.
         assert_eq!(self.stake_accounts_balance, Lamports(0));
         Ok(())
     }
+    pub fn show_removed_error_msg(error: Result<(), LidoError>) -> Result<(), LidoError> {
```

Comment:
It would be a bit simpler if we don’t return the same error again, we can just borrow it.